integration/hive/src/test/java/org/apache/carbondata/hive/TestCarbonHiveRecordReader.java

```diff
@@ -0,0 +1,232 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.hive;
+
+import junit.framework.TestCase;
+import mockit.Mock;
+import mockit.MockUp;
+import org.apache.carbondata.common.CarbonIterator;
+import org.apache.carbondata.core.metadata.AbsoluteTableIdentifier;
+import org.apache.carbondata.core.metadata.CarbonTableIdentifier;
+import org.apache.carbondata.core.metadata.ColumnarFormatVersion;
+import org.apache.carbondata.core.scan.executor.exception.QueryExecutionException;
+import org.apache.carbondata.core.scan.executor.impl.DetailQueryExecutor;
+import org.apache.carbondata.core.scan.executor.infos.BlockExecutionInfo;
+import org.apache.carbondata.core.scan.model.QueryModel;
+import org.apache.carbondata.core.scan.result.BatchResult;
+import org.apache.carbondata.core.scan.result.iterator.AbstractDetailQueryResultIterator;
+import org.apache.carbondata.core.scan.result.iterator.ChunkRowIterator;
+import org.apache.carbondata.core.scan.result.iterator.DetailQueryResultIterator;
+import org.apache.hadoop.hive.serde.serdeConstants;
+import org.apache.hadoop.io.IntWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapred.InputSplit;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.io.ArrayWritable;
+import org.apache.carbondata.hadoop.readsupport.CarbonReadSupport;
+import org.apache.hadoop.fs.Path;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ForkJoinPool;
+
+public class TestCarbonHiveRecordReader extends TestCase {
+    private CarbonHiveRecordReader carbonHiveRecordReaderObj;
+    private AbsoluteTableIdentifier absoluteTableIdentifier ;
+    private QueryModel queryModel = new QueryModel();
+    private CarbonReadSupport<ArrayWritable> readSupport = new CarbonDictionaryDecodeReadSupport<>();
+    private JobConf jobConf = new JobConf();
+    private InputSplit inputSplitNotInstanceOfHiveInputSplit, inputSplitInstanceOfHiveInputSplit;
+    private BatchResult batchResult = new BatchResult();
+    private Writable writable;
+    private CarbonIterator carbonIteratorObject ;
+
+    @BeforeClass
+    public void setUp() throws Exception{
+        String array[] = {"neha","01","vaishali"};
+        writable  = new ArrayWritable(array);
+        absoluteTableIdentifier = new AbsoluteTableIdentifier("/home/neha/Projects/incubator-carbondata/examples/spark2/target/store",
```

Comment:
If I don't have this path, is it ok?

Reply:
@cenyuhai, yes, this path is just used for mocking.